### PR TITLE
[skip ci] Add basic IQ handler tutorial

### PR DESCRIPTION
**Pull request by zofpolkowska**
This PR replaces #2006

### CODE REVIEW

File: doc/developers-guide/Basic-iq-handler.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+%% gen_mod callbacks
+-export([start/2, stop/1]).
+
+%% IQ handlers
+-export([process_iq/4]).
+
+start(Host, _Opts) ->
+    gen_iq_handler:add_iq_handler(ejabberd_sm, Host, <<"erlang-solutions.com:example">>,
+                                  ?MODULE, process_iq, no_queue).
```
Comment (arkgil) [inline doc/developers-guide/Basic-iq-handler.md]: Would it make sense to describe possible options here? I mean `no_queue`, `one_queue` etc. Unless they are already documented somewhere else or it's out of scope of this doc :)

Comment (zofpolkowska) [reply doc/developers-guide/Basic-iq-handler.md]: I haven't found such a description, so I think it would :)

**Comment by mongoose-im** [comment]
[5351.1](https://travis-ci.org/esl/MongooseIM/jobs/413077604) / Erlang 19.3 / small_tests / 30a693c2e07120673b6b42e91e59e680bd826d46
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2021/5351/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2021/5351/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-08-07_12.27.54/lib.mongooseim.logs/run.2018-08-07_12.27.55/suite.log.html)

File: doc/developers-guide/Basic-iq-handler.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                [IQ, IQRes]),
+    {Acc, IQRes}.
+```
+
+
+### IQ processing policies
+
+The server may use one of the following strategies to handle incoming stanzas:
+
+* `no_queue` registers the handler without spawning a new process
```
Comment (erszcz) [inline doc/developers-guide/Basic-iq-handler.md]: Reading it again, I think this description is a bit misleading. I'd use:

> `no_queue` registers a new IQ handler, which will be called in the context of a process serving the connection on which the IQ arrives

**Comment by erszcz** [comment]
I don't have any more comments - it looks good. @arkgil, @fenek?

**Pull request merged**